iotedged/src/main.rs

```diff
@@ -378,6 +381,18 @@ struct Options {
 	#[structopt(long)]
 	device_id: String,
 
+	/// `keyd` API endpoint
+	#[structopt(long, default_value = "unix:///var/lib/aziot/keyd.sock")]
+	keyd_url: url::Url,
+
+	/// `certd` API endpoint
+	#[structopt(long, default_value = "unix:///var/lib/aziot/certd.sock")]
+	certd_url: url::Url,
+
+	/// `identityd` API endpoint
```

Comment:
Would be useful to add a note here that it has to be an `http://` URI. Not just because it gets used with `reqwest`, but because the URL concatenation on line 339 etc only works for URIs handled by `hyper::HttpConnector`.

Reply:
Yeah, I considered adding that comment, but reconsidered given that it should be able to run over UDS at some point anyways. I'll add it in though.

Also, AFAIK, `Url::join` works fine regardless of the underlying Connector (the method is implemented on `Url` directly, and isn't generic .

Reply:
>Also, AFAIK, `Url::join` works fine regardless of the underlying Connector (the method is implemented on `Url` directly, and isn't generic.

No, my point is the unix connector needs to use `http://foo/rest/of/the/uri` as opposed to `unix:///path/to/socket/rest/of/the/uri`, as you can see in https://github.com/Azure/iot-identity-service/blob/fe4a7a7fb5378558a99469150fb8b2c0ca5ba913/key/aziot-key-client-async/src/lib.rs#L41

Reply:
uhhh, not gonna lie, I'm not _entirely_ sure what's going on here, but I'll take your word for it.

Reply:
`unix:///path/to/socket/rest/of/the/uri` is ambiguous, yes? There is a need to be able to know what the socket path is and what the URI is.

`hyperlocal`'s connector encodes the socket path in the hostname by converting it to hex, but the downside is you need to know it's being used because you have to use its special function to construct the URI with the encoded hostname. So you can't just concatenate the base URI with the request URI path.

This repo uses its own connector instead of `hyperlocal` that takes in the socket path at construction time instead, so it doesn't need the hostname to tell it what the socket path is. So the actual value of the hostname is irrelevant and can just be `foo`. But still, this too doesn't let you just concatenate the base URI with the request URI path.

Reply:
Ahhhh, gotcha. I didn't know that. Makes sense though.